### PR TITLE
Update Examples to Adhere to Naming Conventions

### PR DESCRIPTION
**Pull request by maddyb99**
# Status
**READY**

## Breaking Changes
NO, the changes introduced do not break the project.

## Description
This PR addresses Issue #1024
Currently I have refactored three examples, namely, flutter_timer, flutter_firebase_login, flutter_bloc_with_stream. I have followed the official naming conventions as strictly as possible.

This PR contains all the changes in the last PR and incorporated all the changes you suggestes. I've even started refactoring other examples. I'll push them when all are completed.

Thank you for being patient with me as this is the first time I'm trying to contribute to other projects. I've fixed the squash I did by mistake. 

## Related PRs
List related PRs against other branches:

branch | PR
#1124  | [link](https://github.com/felangel/bloc/pull/1124)

## Todos
None

## Steps to Test or Reproduce
NA

## Impact to Remaining Code Base
This PR will affect:
This will not affect the working of any other part of the project

*

### CODE REVIEW

File: examples/flutter_firebase_login/lib/authentication_bloc/authentication_bloc.dart [hunk 1]
```diff
@@ -16,37 +16,37 @@ class AuthenticationBloc
...
-    } else if (event is LoggedIn) {
-      yield* _mapLoggedInToState();
-    } else if (event is LoggedOut) {
-      yield* _mapLoggedOutToState();
+    if (event is AuthenticationStarted) {
+      yield* _mapAuthenticationStartedToState();
+    } else if (event is AuthenticationLoggedIn) {
+      yield* _mapAuthenticationLoggedInToState();
+    } else if (event is AuthenticationLoggedOut) {
+      yield* _mapAuthenticationLoggedOutToState();
```
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/authentication_bloc/authentication_bloc.dart:30]: Updated all helper functions acrross all examples

File: examples/flutter_firebase_login/lib/login/bloc/login_bloc.dart [hunk 2]
```diff
@@ -36,10 +36,10 @@ class LoginBloc extends Bloc<LoginEvent, LoginState> {
...
   @override
   Stream<LoginState> mapEventToState(LoginEvent event) async* {
-    if (event is EmailChanged) {
-      yield* _mapEmailChangedToState(event.email);
-    } else if (event is PasswordChanged) {
-      yield* _mapPasswordChangedToState(event.password);
+    if (event is LoginEmailChanged) {
+      yield* _mapLoginEmailChangedToState(event.email);
+    } else if (event is LoginPasswordChanged) {
+      yield* _mapLoginPasswordChangedToState(event.password);
```
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/login/bloc/login_bloc.dart:42]: Updated all helper functions

File: examples/flutter_firebase_login/lib/login/bloc/login_event.dart [hunk 3]
```diff
@@ -32,24 +32,6 @@ class PasswordChanged extends LoginEvent {
...
-
-  @override
-  List<Object> get props => [email, password];
-
-  @override
-  String toString() {
-    return 'Submitted { email: $email, password: $password }';
-  }
-}
-
```
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/login/bloc/login_event.dart:52]: Removed un-used class

File: examples/flutter_firebase_login/lib/register/bloc/register_bloc.dart [hunk 4]
```diff
@@ -37,28 +39,29 @@ class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
...
-    } else if (event is PasswordChanged) {
-      yield* _mapPasswordChangedToState(event.password);
-    } else if (event is Submitted) {
-      yield* _mapFormSubmittedToState(event.email, event.password);
+    if (event is RegisterEmailChanged) {
+      yield* _mapRegisterEmailChangedToState(event.email);
+    } else if (event is RegisterPasswordChanged) {
+      yield* _mapRegisterPasswordChangedToState(event.password);
+    } else if (event is RegisterSubmitted) {
+      yield* _mapRegisterSubmittedToState(event.email, event.password);
```
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/register/bloc/register_bloc.dart:47]: Updated all helper functions

File: examples/flutter_timer/lib/bloc/timer_bloc.dart [hunk 5]
```diff
@@ -30,16 +30,16 @@ class TimerBloc extends Bloc<TimerEvent, TimerState> {
...
+    if (event is TimerStarted) {
+      yield* _mapTimerStartedToState(event);
+    } else if (event is TimerPaused) {
+      yield* _mapTimerPausedToState(event);
+    } else if (event is TimerResumed) {
+      yield* _mapTimerResumedToState(event);
+    } else if (event is TimerReset) {
+      yield* _mapTimerResetToState(event);
+    } else if (event is TimerTicked) {
+      yield* _mapTimerTickedToState(event);
```
Comment (maddyb99) [inline examples/flutter_timer/lib/bloc/timer_bloc.dart:42]: Updated all helper functions

File: examples/flutter_firebase_login/lib/authentication_bloc/authentication_bloc.dart (hunk 1 above)
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/authentication_bloc/authentication_bloc.dart:30]: Updated all helper functions acrross all examples

File: examples/flutter_firebase_login/lib/login/bloc/login_bloc.dart (hunk 2 above)
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/login/bloc/login_bloc.dart:42]: Updated all helper functions

File: examples/flutter_firebase_login/lib/register/bloc/register_bloc.dart (hunk 4 above)
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/register/bloc/register_bloc.dart:47]: Updated all helper functions

File: examples/flutter_timer/lib/bloc/timer_bloc.dart (hunk 5 above)
Comment (maddyb99) [inline examples/flutter_timer/lib/bloc/timer_bloc.dart:42]: Updated all helper functions

File: examples/flutter_firebase_login/lib/login/bloc/login_event.dart (hunk 3 above)
Comment (maddyb99) [inline examples/flutter_firebase_login/lib/login/bloc/login_event.dart:52]: Removed un-used class

**Comment by maddyb99** [comment]
So I figured out how to fix the issue I had faced in the last pull request and to some extent, replicated in this PR. I also figured out why it happened in the first place. Future PRs will be clean with only 1 commit and free of bloat commits and comments 😊

**Comment by maddyb99** [comment]
Updated my branch with latest commits.
Refactored 3 more examples according to naming convention: flutter_login, flutter_infinite_list, flutter_dynamic_form

File: examples/flutter_login/test/unit/login/login_event_test.dart [hunk 6]
```diff
@@ -12,7 +12,10 @@ void main() {
       });
 
       test(
-          'toString is LoginButtonPressed { username: username, password: password }',
+          //removed 'toString' as it would fail command:
+          // flutter format --set-exit-if-changed test
+          //line length exceeded 80 characters
+          'is LoginButtonPressed { username: username, password: password }',
```
Comment (maddyb99) [inline examples/flutter_login/test/unit/login/login_event_test.dart:18]: @felangel , I would like to request you to look into this issue

Comment (felangel) [reply examples/flutter_login/test/unit/login/login_event_test.dart:18]: will push a fix in a bit 👍

Comment (maddyb99) [reply examples/flutter_login/test/unit/login/login_event_test.dart:18]: One possible solution:
`flutter format -n --set-exit-if-changed -- --line-length 140`
Reference: [https://github.com/dart-lang/sdk/issues/37657](https://github.com/dart-lang/sdk/issues/37657)

Comment (felangel) [reply examples/flutter_login/test/unit/login/login_event_test.dart:18]: can we revert this and use multi-line strings?

```dart
'toString is '
'LoginButtonPressed '
'{ username: username, password: password }'
```

Comment (maddyb99) [reply examples/flutter_login/test/unit/login/login_event_test.dart:18]: Well, I'll do that.
Also, I will refactor other examples, as soon as I'm free from my college assignments :smile:
Please review these 6 examples for the time being.
Thanks!

Comment (maddyb99) [reply examples/flutter_login/test/unit/login/login_event_test.dart]: Fixed this and also updated my refs with your master branch.

File: examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart [hunk 7]
```diff
@@ -71,7 +74,8 @@ class NewCarBloc extends Bloc<NewCarEvent, NewCarState> {
     );
   }
 
-  Stream<NewCarState> _mapYearChangedToState(YearChanged event) async* {
+  Stream<NewCarState> _mapNewCarYearChangedToState(
+      NewCarYearChanged event) async* {
```
Comment (felangel) [inline examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart:78]:
```suggestion
      NewCarYearChanged event,) async* {
```

File: examples/flutter_login/test/unit/authentication/authentication_event_test.dart [hunk 8]
```diff
@@ -3,36 +3,36 @@ import 'package:flutter_test/flutter_test.dart';
...
       });
 
-      test('toString is "AppStarted"', () {
-        expect(AppStarted().toString(), 'AppStarted');
+      test('toString is "AuthenticationStarted"', () {
+        expect(AuthenticationStarted().toString(), 'AuthenticationStarted');
       });
     });
 
     group('LoggedIn', () {
```
Comment (felangel) [inline examples/flutter_login/test/unit/authentication/authentication_event_test.dart:16]:
```suggestion
    group('AuthenticationLoggedIn', () {
```

**Comment by felangel** [comment]
@maddyb99 thanks so much for the amazing contribution! I'm just going to do one more pass over this PR tomorrow morning and should be good to merge 🎉 
Sorry for the delay and thanks again 💯

File: examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart (hunk 7 above)
Comment (maddyb99) [reply examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart:78]: should I refactor to add trailing commas throughout examples or are you suggesting just this one?

**Comment by maddyb99** [comment]
Thank you @felangel . It was great to help out while learning something new here 🙇

File: examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart (hunk 7 above)
Comment (felangel) [reply examples/flutter_dynamic_form/lib/bloc/new_car_bloc.dart:78]: I wouldn’t worry about it actually

**Pull request merged**